sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppBasicImpl.java

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.appservice.implementation;
+
+import com.azure.resourcemanager.appservice.AppServiceManager;
+import com.azure.resourcemanager.appservice.fluent.inner.SiteInner;
+import com.azure.resourcemanager.appservice.models.FunctionApp;
+import com.azure.resourcemanager.appservice.models.FunctionAppBasic;
+import com.azure.resourcemanager.resources.fluentcore.arm.models.HasManager;
+import reactor.core.publisher.Mono;
+
+class FunctionAppBasicImpl extends WebSiteBaseImpl implements FunctionAppBasic, HasManager<AppServiceManager> {
+
+    private final AppServiceManager myManager;
+
+    FunctionAppBasicImpl(SiteInner innerObject, AppServiceManager myManager) {
+        super(innerObject);
+        this.myManager = myManager;
+    }
+
+    @Override
+    public FunctionApp refresh() {
+        return this.refreshAsync().block();
+    }
+
+    @Override
+    public Mono<FunctionApp> refreshAsync() {
```

Comment:
just curious, why name it refresh on a get call?

Reply:
To make it like other common resource instance, e.g.`FunctionApp`, which implement a `Refreshable<FunctionApp>`, which `refresh` update itself and return a `FunctionApp` instance (actually just `this`).

And here `FunctionAppBasic` has `refresh` to update itself, and return a more complete `FunctionApp` instance.